### PR TITLE
[charts] Setup the repository for charts-pro

### PR DESCRIPTION
**Pull request by alexfauquette**
This PR is just about the update of the config files. Mostly modifications taken from #12240 and #12610

Followup will be

- Add the watermark on the pro chart container
- Make a PoC of the heatmap

### CODE REVIEW

File: packages/x-charts-pro/.mocharc.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  'watch-ignore': [
+    // default
+    '.git',
+    // node_modules can be nested with workspaces
+    '**/node_modules/**',
+    // Unrelated directories with a large number of files
+    '**/build/**',
+    'docs/.next/**',
+  ],
+  spec: ['packages/x-charts/**/*.test.{js,ts,tsx}'],
```
Comment (flaviendelangle) [inline packages/x-charts-pro/.mocharc.js:33]: Should we register the `x-charts-pro` test files here instead?

File: packages/x-charts-pro/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { LicenseInfo as LicenseInfoExport } from '@mui/x-license';
+/**
+ * @deprecated Use `@mui/x-license` package instead:
+ * @example import { LicenseInfo } from '@mui/x-license';
+ */
+export class LicenseInfo extends LicenseInfoExport {}
```
Comment (flaviendelangle) [inline packages/x-charts-pro/src/index.ts:6]: Do we really want to add this if its deprecated?
We don't have it on Tree View, we should of course be consistent across the 2 packages whatever we end up deciding.

Comment (alexfauquette) [reply packages/x-charts-pro/src/index.ts:6]: I never touched to those license files. I just copy pasted those lines from the date picker pro

Comment (flaviendelangle) [reply packages/x-charts-pro/src/index.ts]: In v8 we will remove those deprecated exports from pickers and grid
And everybody will just import from the license package

**Review by flaviendelangle — APPROVED** [review]
Looks good :ok_hand:

**Pull request merged**